src/components/sidebar/index.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState } from 'react';
+import { styled } from '@mui/material/styles';
+import {
+  Box,
+  Drawer,
+  CssBaseline,
+  List,
+  Divider,
+  IconButton,
+  ListItem,
+  ListItemButton,
+  ListItemIcon,
+  ListItemText,
+} from '@mui/material';
+import { Menu, ChevronLeft } from '@mui/icons-material';
+import {
+  GridOnOutlined,
+  DescriptionOutlined,
+  PlagiarismOutlined,
+  LaptopWindowsOutlined,
+  ScreenSearchDesktopOutlined,
+  SettingsOutlined,
+  PersonOutlineOutlined,
+} from '@mui/icons-material';
+
+const drawerWidth = 240;
+
+const DrawerHeader = styled('div')(({ theme }) => ({
+  display: 'flex',
+  alignItems: 'center',
+  padding: theme.spacing(0, 1),
+  ...theme.mixins.toolbar,
+  justifyContent: 'flex-end',
+}));
+
+const sidebarButtonsList = [
```

Comment:
PP: constants should always be uppercase and use underlines naming: "SIDEBAR_BUTTONS_LIST".
Also, I would extract this in a separate file.